src/stores/params.js

```diff
@@ -211,6 +213,150 @@ export class DataFetcher {
     return r;
   }
 
+  /**
+   * @param {(Sensor|SensorParam)[]} sensors
+   * @param {Region|RegionParam} region
+   * @param {TimeFrame} timeFrame
+   * @return {Promise<EpiDataRow[]>[]}
+   */
+  fetchNSensor1RegionNDates(sensors, region, timeFrame) {
+    sensors = sensors.map((sensor) => (sensor instanceof SensorParam ? sensor.value : sensor));
+    region = region instanceof RegionParam ? region.value : region;
+
+    const missingDataSensors = sensors.filter((sensor) => !this.cache.has(this.toWindowKey(sensor, region, timeFrame)));
+    // we can only fetch the same data source for now
+    const dataSources = new Set(missingDataSensors.map((d) => d.id));
+
+    if (dataSources.size === 1 && timeFrame.range !== ALL_TIME_FRAME.range) {
+      // fill up the cache
+      const expectedDays = timeFrame.difference;
+      const batchSize = Math.floor(MAX_DATA_ROWS / expectedDays);
+      // console.log(batchSize);
+      for (let i = 0; i < missingDataSensors.length; i += batchSize) {
+        const slice = missingDataSensors.slice(i, Math.min(missingDataSensors.length, i + batchSize));
+        const sliceSensor = {
+          ...missingDataSensors[0],
+          signal: slice.map((d) => d.signal).join(','),
+        };
+        const data = fetchData(
+          sliceSensor,
+          region.level,
+          region.propertyId,
+          timeFrame.range,
+          {},
+          {
+            multiValues: false,
+            transferSignal: true,
+            factor: sensorFactor(sliceSensor),
+          },
+        ).then(addNameInfos);
+        for (const s of slice) {
+          // compute slice per sensor and fill the cache
+          const sensorData = data
+            .then((rows) => rows.filter((d) => d.signal === s.signal))
+            .then((rows) => addMissing(rows, s));
+          this.cache.set(this.toWindowKey(s, region, timeFrame), sensorData);
+        }
+      }
+    }
+
+    // use cached version
+    return sensors.map((sensor) => this.fetch1Sensor1RegionNDates(sensor, region, timeFrame));
+  }
+
+  /**
+   * @param {(Sensor|SensorParam)[]} sensors
+   * @param {Region[]} regions
+   * @param {TimeFrame} timeFrame
+   * @return {Promise<EpiDataRow[]>[]}
+   */
+  fetchNSensorNRegionNDates(sensors, regions, timeFrame) {
```

Comment:
These fetch methods read pretty well.  If we were to migrate some of this logic into the API layer, how much of an improvement would that be in load time and reducing front-end complexity?

Reply:
the following features for the api will make things simpler
 * possibility to fetch multiple signal at a time (so far only possible for the same data source)
 * no more max rows = no batching needed => will be part of flask implementation
 * multiple geo types at once (there is an issue for that), e..g nation and state at the same time
 * min/mean/max per given region / date frame -> easier trend computation